### PR TITLE
Allow to use custom http modules(e.g unit-http)

### PR DESCRIPTION
**Pull request by alg3rd**
This change will allow to use application server Nginx Unit(https://unit.nginx.org/installation/#installation-nodejs-package) in koa applications without any changes to existing codebases.

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -61,7 +61,12 @@ module.exports = class Application extends Emitter {
 
   listen(...args) {
     debug('listen');
-    const server = http.createServer(this.callback());
+    let httpModule = http;
+    if (args.length > 0 && typeof args[0] === 'object') {
+      httpModule = args[0];
```
Comment (kroleg) [inline lib/application.js]: you can write `httpModule = args.shift();`

**Comment by kroleg** [comment]
i don't think that this should be included in koa because there is already an easy way to use custom http server:
```
const app = new Koa();
const unitHttp = require('unit-http');
const server = unitHttp.createServer(app.callback());
server.listen(port, () => {
    console.log('Listening on port ' + port);
});
```

**Comment by fl0w** [comment]
Aye, this is a no go from me. Use `app.callback() // e.g. spdy.listen(app.callback())`

**Comment by alg3rd** [comment]
Thank you for responses.

**Pull request closed without merging**